ckanext/validation/model.py

```diff
@@ -17,26 +14,35 @@ def make_uuid():
     return str(uuid.uuid4())
 
 
-Base = declarative_base(metadata=metadata)
+class Validation(model.DomainObject):
+    @classmethod
+    def get(cls, **kw):
+        '''Finds all the instances required.'''
+        query = model.Session.query(cls).autoflush(False)
+        return query.filter_by(**kw).all()
 
 
-class Validation(Base):
-    __tablename__ = u'validation'
+validation_table = sa.Table('validation', metadata,
+                            sa.Column('id', sa.types.UnicodeText, primary_key=True, default=make_uuid),
+                            sa.Column('resource_id', sa.types.UnicodeText, primary_key=False),
+                            sa.Column('status', sa.types.UnicodeText, primary_key=False, default='created'),
+                            sa.Column('created', sa.types.DateTime, primary_key=False, default=datetime.datetime.utcnow),
+                            sa.Column('finished', sa.types.DateTime, primary_key=False),
+                            sa.Column('report', JSON, primary_key=False),
+                            sa.Column('error', JSON, primary_key=False)
+                            )
 
-    id = Column(Unicode, primary_key=True, default=make_uuid)
-    resource_id = Column(Unicode)
-    status = Column(Unicode, default=u'created')
-    created = Column(DateTime, default=datetime.datetime.utcnow)
-    finished = Column(DateTime)
-    report = Column(JSON)
-    error = Column(JSON)
+mapper(Validation, validation_table)
 
 
 def create_tables():
-    Validation.__table__.create()
+    metadata.create_all(model.meta.engine)
 
     log.info(u'Validation database tables created')
 
 
 def tables_exist():
-    return Validation.__table__.exists()
+    eng = sa.create_engine(config.get('sqlalchemy.url'))
```

Comment:
This is a very inelegant approach. The `create_tables` function is idempotent so this function is not really needed anyway.